Source/Scene/Label.js

```diff
@@ -279,6 +290,15 @@ define([
                 }
                 //>>includeEnd('debug');
 
+                if (Label.enableRightToLeftDetection) {
+                    if (this._originalValue === value) {
+                        value = this._text;
+                        return;
+                    }
+                    this._originalValue = value;
+                    value = reverseRtl(value);
```

Comment:
I just noticed that we are mutating the value supplied by the user so that the getter is actually the reversed text.  I think this is undesired behavior.  With few exceptions, a simple setter/getter should always return the value assigned to it.  I can submit a change to prevent this, but if you feel strongly about it, please let me know.